src/raftstore/store/worker/apply.rs

```diff
@@ -195,48 +194,126 @@ impl ApplyCallback {
 
 struct ApplyContext<'a> {
     host: &'a CoprocessorHost,
-    wb: WriteBatch,
+    wb: Option<WriteBatch>,
     cbs: MustConsumeVec<ApplyCallback>,
+    apply_res: Vec<ApplyRes>,
     wb_last_bytes: u64,
     wb_last_keys: u64,
+    committed_count: usize,
+    sync_log_hint: bool,
     sync_log: bool,
     exec_ctx: Option<ExecContext>,
     use_delete_range: bool,
 }
 
 impl<'a> ApplyContext<'a> {
-    fn new(host: &CoprocessorHost, use_delete_range: bool) -> ApplyContext {
+    fn new(host: &CoprocessorHost) -> ApplyContext {
         ApplyContext {
             host: host,
-            wb: WriteBatch::with_capacity(DEFAULT_APPLY_WB_SIZE),
+            wb: None,
             cbs: MustConsumeVec::new("callback of apply context"),
+            apply_res: vec![],
             wb_last_bytes: 0,
             wb_last_keys: 0,
+            committed_count: 0,
+            sync_log_hint: false,
```

Comment:
It is easy to confuse with `sync_log_hint` and `sync_log`.

Reply:
Maybe we should rename `sync_log_hint` as `enable_sync_log` and `sync_log` as `should_sync_log`.